readme.md

```diff
@@ -0,0 +1,101 @@
+# Spring Boot EMail Service
+
+A reusable mailer service to send emails.
+
+## Setup
+
+Add the mailer service to the classpath by adding the following maven dependency. Versions haven be found [here](../../packages)
+
+```xml
+
+<dependency>
+    <groupId>it.aboutbits.springboot</groupId>
+    <artifactId>emailservice</artifactId>
+    <version>x.x.x</version>
+</dependency>
+```
+
+### Attachments
+
+If you want to use attachments, you will have to create a bean implementing this interface: [AttachmentDataSource.java](src%2Fmain%2Fjava%2Fit%2Faboutbits%2Fspringboot%2Femailservice%2Flib%2FAttachmentDataSource.java)  
+This step is optional.
+
+## Usage
+
+### Sending an Email
+
+Use the `ManageEmail` service to schedule an email.
+
+#### Example
+
+```java
+// @formatter:off
+public void sendMail(final String to,final String subject,final String htmlBody,final String plainTextBody) {
+        manageEmail.schedule(
+                EmailParameter.builder()
+                        .scheduleAt(OffsetDateTime.now())
+                        .email(EmailParameter.Email.builder()
+                                .fromName(fromName)
+                                .fromAddress(fromAddress)
+                                .recipient(to)
+                                .subject(subject)
+                                .textBody(plainTextBody)
+                                .htmlBody(htmlBody)
+                                .build())
+                        .build()
+        );
+}
+// @formatter:on 
+```
+
+### Querying Emails
+
+To read email datasets from the database use this class: [QueryEmail.java](src%2Fmain%2Fjava%2Fit%2Faboutbits%2Fspringboot%2Femailservice%2Flib%2Fapplication%2FQueryEmail.java)
+
+### Reporting callback
+
+If you want to receive a report after each run of the scheduler, create a Bean implementing [EmailSchedulerCallback.java](src%2Fmain%2Fjava%2Fit%2Faboutbits%2Fspringboot%2Femailservice%2Flib%2FEmailSchedulerCallback.java)
+
+### Configuration
+
+To enable this service just add `@EnableEmailService` to your main class. You must also enable `@EnableScheduling` to allow the email queue to be processed.
+
+```java
+
+@SpringBootApplication
+@EnableEmailService
+@EnableScheduling
+public class App {
+    public static void main(final String[] args) {
+        SpringApplication.run(App.class, args);
+    }
+}
+```
+
+The following configuration options are available:
+
+| Name                                   | Default     | Description                                                           |
+|----------------------------------------|-------------|-----------------------------------------------------------------------|
+| `lib.emailservice.migrations.enabled`  | true        | Enables database migrations.                                          |
```

Comment:
wdyt about namespacing it with `aboutbits`.

`aboutbits.emailservice.migrations.enabled`